sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -0,0 +1,435 @@
+      Integer ttl,
+      ExecutionContext context) {
+    // pre-process to support array elements
+    final List<String> subPathProcessed = processSubDocumentPath(subPath);
+    // read the root
+    final JsonNode root = readPayload(payload);
+    // explicitly forbid arrays and empty objects
+    if (root.isArray()) {
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_PATCH_ARRAY_NOT_ACCEPTED);
+    }
+    if (root.isObject() && root.isEmpty()) {
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_PATCH_EMPTY_NOT_ACCEPTED);
+    }
+
+    return jsonSchemaManager
+        .validateJsonDocument(
+            tableManager.getValidCollectionTable(namespace, collection),
+            root,
+            !subPathProcessed.isEmpty())
+        .onItem()
+        .transform(
+            __ -> {
+              // shred rows
+              List<JsonShreddedRow> rows = documentShredder.shred(root, subPathProcessed);
+
+              // call patch document
+              return writeBridgeService.patchDocument(
+                  namespace, collection, documentId, subPathProcessed, rows, ttl, context);
+            })
+        .map(any -> new DocumentResponseWrapper<>(documentId, null, null, context.toProfile()));
+  }
+
+  /**
+   * Deletes a document with given ID in the given namespace and collection at the specified
+   * sub-path. Any previously existing sub-document at the given path will be removed.
+   *
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to delete
+   * @param subPath Sub-path of the document to delete. If empty will delete the whole doc.
+   * @param context Execution content
+   * @return Flag representing if the operation was success.
+   */
+  public Uni<Boolean> deleteDocument(
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      ExecutionContext context) {
+    // pre-process to support array elements
+    List<String> subPathProcessed = processSubDocumentPath(subPath);
+    return writeBridgeService
+        .deleteDocument(namespace, collection, documentId, subPathProcessed, context)
+        .onItem()
+        .transform(__ -> true);
+  }
+
+  // we need to transform the stuff to support array elements
+  private List<String> processSubDocumentPath(List<String> subDocumentPath) {
+    return subDocumentPath.stream()
+        .map(path -> DocsApiUtils.convertArrayPath(path, configuration.maxArrayLength()))
+        .collect(Collectors.toList());
+  }
+
+  private Uni<Integer> determineTtl(
+      String namespace, String collection, String documentId, ExecutionContext ctx) {
+    return readBridgeService
+        .getDocumentTtlInfo(namespace, collection, documentId, ctx)
+        .collect()
+        .asList()
+        .onItem()
+        .transform(
+            rawDocumentList ->
+                rawDocumentList.get(0).rows().get(0).getLong("ttl(leaf)").intValue());
```

Comment:
please make sure that document does exist, if not return `0`.. So when applying the change to uni, please make null check and switch.. see all the checks in V1

```
    return Single.defer(
        () ->
            searchService
                .getDocumentTtlInfo(
                    db.getQueryExecutor(), namespace, collection, documentId, context)
                .singleElement()
                .map(
                    document -> {
                      Integer ttl =
                          document.rows().stream()
                              .map(
                                  row ->
                                      row.getInt(
                                          String.format(
                                              "ttl(%s)", DocsApiConstants.LEAF_COLUMN_NAME)))
                              .filter(Objects::nonNull)
                              .mapToInt(Integer::valueOf)
                              .max()
                              .getAsInt();
                      return ttl == null ? 0 : ttl;
                    })
                .defaultIfEmpty(0));
  }
```